x-pack/plugins/apm/public/components/app/correlations/correlations_table.tsx

```diff
@@ -59,7 +66,28 @@ export function CorrelationsTable<T extends SignificantTerm>({
     [trackApmEvent]
   );
   const history = useHistory();
-  const columns: Array<EuiBasicTableColumn<T>> = [
+
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+
+  const pagination = useMemo(() => {
+    const itemCount = significantTerms?.length ?? 0;
+    return {
+      pageIndex,
+      pageSize,
+      totalItemCount: itemCount,
+      pageSizeOptions: PAGINATION_SIZE_OPTIONS,
+    };
+  }, [pageIndex, pageSize, significantTerms]);
+
+  const onTableChange = useCallback(({ page }) => {
+    const { index, size } = page;
+
+    setPageIndex(index);
+    setPageSize(size);
+  }, []);
```

Comment:
shall we reset a page index in case page size has been changed?

Reply:
EUI table handles this for us on page size change